src/libbson/src/bson/bson-json.c

```diff
@@ -833,8 +835,29 @@ _bson_json_read_int64_or_set_error (bson_json_reader_t *reader, /* IN */
    return true;
 }
 
+static bool
+_unhexlify_uuid (const char *uuid, uint8_t *out, size_t max)
```

Comment:
Shamelessly copied this from `corpus_test_unhexlify`. Let me know if this functionality should be extracted. I've adapted the code to work directly on the BSON buffer but can refactor this to allocate a buffer of the correct size, as is done in the original method.

If there's value in having this available in general, I can also extract this (e.g. to `bson_uuid_pton` to be consistent with the `bson_b64_*` methods.